src/main/java/org/tron/common/runtime/vm/program/InternalTransaction.java

```diff
@@ -21,22 +21,43 @@
 import static org.apache.commons.lang3.ArrayUtils.nullToEmpty;
 import static org.tron.common.utils.ByteUtil.toHexString;
 
+import com.google.common.primitives.Longs;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
+import java.util.Arrays;
 import org.tron.common.crypto.ECKey;
+import org.tron.common.crypto.ECKey.ECDSASignature;
+import org.tron.common.crypto.Hash;
 import org.tron.common.runtime.vm.DataWord;
 import org.tron.common.utils.ByteUtil;
+import org.tron.core.capsule.TransactionCapsule;
 import org.tron.protos.Protocol.Transaction;
 
-//import org.tron.common.utils.RLP;
-
-public class InternalTransaction extends VMTransaction {
+public class InternalTransaction {
 
+  public Transaction transaction;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Make transaction a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1220&issues=AWT01HPBvlEfaeH0aL_c&open=AWT01HPBvlEfaeH0aL_c)

[ISSUE_KEY]: AWT01HPBvlEfaeH0aL_c